### PR TITLE
Support additional shebang-like lines

### LINKED ISSUE #5: Shebabng not start with '#!'
Shebang of zsh completion files starts with `#compdef` which is not recognized.
https://github.com/davidhewitt/shebang-language-associator/blob/27492a949540388fb2fc388e3915c1239abacb6b/src/extension.ts#L53-L55

### PR DESCRIPTION
**Pull request by Adelrisk**
This will resolve #5 and also add support for (go and jbang) scripts: https://unix.stackexchange.com/a/162535

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -48,9 +59,10 @@ function checkFile(doc: vscode.TextDocument) {
     let shebang = doc.lineAt(0);
 
     /*
-     * Do nothing if the first line is not a shebang line.
+     * Do nothing if the first line is not a shebang-like line.
      */
-    if (!shebang.text.startsWith("#!")) {
+    const match = acceptedPrefixes.find(prefix => shebang.text.startsWith(prefix));
+    if (!match) {
```
Comment (davidhewitt) [inline src/extension.ts]: I think `Array.some` might be a better fit here?

```suggestion
    const match = acceptedPrefixes.some(prefix => shebang.text.startsWith(prefix));
    if (!match) {
```

**Review by davidhewitt — COMMENTED** [review]
Thanks, just one quick implementation nit!

Would you also be willing to add a CHANGELOG entry please (will release this as 1.2.0)? Then when I can find a spare moment, I'll release 👍

**Comment by Adelrisk** [comment]
> Thanks, just one quick implementation nit!

Not a problem, that one should be obvious. I'm still working out why it didn't occur to me.
 
> Would you also be willing to add a CHANGELOG entry please (will release this as 1.2.0)? Then when I can find a spare moment, I'll release +1

Done.

**Review by davidhewitt — APPROVED** [review]
Thanks!

**Pull request merged**